File: src/app/models/product.ts
export interface Product {
  productId: number;
  name: string;
  description: string;
  price: number;
  imageUrl: string;
  stock: number;
}

export interface ProductDTO {
  name: string;
  description: string;
  price: number;
  imageUrl: string;
  stock: number;
}

File: WebApplication17/Models/Product.cs
using System.ComponentModel.DataAnnotations;


namespace WebApplication17.Models
{
    public class Product
    {

        public int ProductId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Description { get; set; }

        [Range(1, 1000000)]
        public decimal Price { get; set; }

        public string ImageUrl { get; set; }

        [Range(0, int.MaxValue)]
        public int Stock { get; set; }

       
        public ICollection<OrderItem> OrderItems { get; set; }

    }
}
